Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1566,6 +1566,8 @@ bool RenderElement::isInsideEntirelyHiddenLayer() const
 {
     if (isSVGLayerAwareRenderer() && document().settings().layerBasedSVGEngineEnabled() && enclosingLayer()->enclosingSVGHiddenOrResourceContainer())
         return true;
+    if (!enclosingLayer())
```

Comment:
Let's not compute `enclosingLayer()` twice. Write it as:

```
if (style().usedVisibility() != Visibility::Visible)
  return true;

CheckedPtr layer  = enclosingLayer();
if (!layer)
  return true;

return !layer->hasVisibleContent();
```

Reply:
I used the stack variable also on the first check.